mrobosub_gnc/mrobosub_gnc/pid_dof_controller.py

```diff
@@ -61,37 +55,82 @@ def __init__(self, dof_name: str):
             Float64, f"/target_twist/{dof_name}", self.target_twist, qos_profile=1
         )
 
+    def set_params(self, _params = None):    
+        self.kp = self.get_parameter('kp').get_parameter_value().double_value
+        self.kd = self.get_parameter('kd').get_parameter_value().double_value
+        self.ki = self.get_parameter('ki').get_parameter_value().double_value
+        self.angular = self.get_parameter('angular').value
+        self.integral_windup_limit = self.get_parameter('integral_windup_limit').get_parameter_value().double_value # absolute value of max integral accumulated
+        # integral term does not go outside [ - self.integral_windup_limit, self.integral_windup_limit ]
+
+    def declare_params(self):
+        self.declare_parameter("kp", 1.0)
+        self.declare_parameter("kd", 1.0)
+        self.declare_parameter("ki", 1.0)
+        self.declare_parameter("angular", False)
+        self.declare_parameter("integral_windup_limit", 1.0)
+        self.set_params()
+
     def target_pose_callback(self, target_pose: Float64):
```

Comment:
If this isn't actually being called between nodes do we need to be using Float64? or can we just use python's `float`